### ISSUE TITLE
Blocking form submissions with blacklisted keywords

### ISSUE THREAD

**Issue by mdee4** [CalderaWP/Caldera-Forms#2836]
I think it would be very helpful to have to the ability to block form submissions based on blacklisted keywords.

The current spam protection is a good start, but spam will always find a way through. To further protect against spam, a keyword blacklist would be useful.

For example, say you get a lot of marketing spam submissions. It would be nice if I could go into Caldera Forms and add the keyword "marketing" to a blacklist. Then, if that word is used in a form, the submission won't send out email notifications. This could potentially save a lot of spam from getting through to me and my clients.

**Comment by Shelob9** [comment]
Thanks for using Caldera Forms. We can not accept bug reports that do not follow our template. I hope you understand that the information that we ask for would explain if this is a bug or a site-specific issue. Right now I have no idea what is going wrong on your site.

Please open a new ticket following the new issue template.

Please note, this is not a support channel.  If you need immediate help with your site or help troubleshooting Caldera Forms on your site, please open a support ticket at https://CalderaForms.com/support

**Comment by mdee4** [comment]
My bad. 

I already contacted Caldera support with this feature request and they told me to post the feature request on this Github account. Then, when I read the Github template here, it says...

"## Is This A Feature Request?
* If so, that's fine, please describe your desired featured and delete the rest of this template."

To me, it sure looks like this is where I'm supposed to leave a feature request. 

Where do I leave a feature request? Caldera support directs me here and now you're directing me to Caldera support.

**Comment by Shelob9** [comment]
Yep. I misunderstood. You're correct this is a feature request. I will keep it in mind. Thanks for the suggestion. It sounds like you used our contact form, not the support form for customers and they correctly sent you here and I misread it.


This is totally doable now with custom code. If you are a paying customer and put in your support request https://calderaforms.com/support that Josh said he'd dig up the code for this, the support team will ask me for that code.

**Comment by Shelob9** [comment]
Reasons why not to so this:

Our anti-spam in CF Pro does the same thing with a smarter algorithm someone else maintains.

This is easy to circumvent.

Reasons to do this:
It let's user add there own rules instead of trusting the 3rd part api's more general solution.

**Comment by mdee4** [comment]
That's awesome! I'm a paid pro user. Just sent a request using the Caldera Forms Pro support form.

I think it'd be a great feature to add. Sometimes spam gets through and it'd be nice to know I could just block certain submissions based on keywords for clients. I know Gravity Forms can do this. A common example being someone who non-stop sends marketing emails to a client manually and it keeps getting through spam filters. A quick keyword block and the spam stops.

**Comment by JeePeeNL** [comment]
I think it's best to use the blacklist that's already build into WordPress so you don't have to maintain two separate lists (nor a separate list per form).
- Admin » Settings » Discussion » Blacklist

**Comment by Shelob9** [comment]
Duplicate of #2880 should get fixed in #3544
